lib/countly-storage.js

```diff
@@ -125,11 +125,22 @@ const fileStorage = {
  * @param {StorageTypes} storageType - Whether to use memory only storage or not
  * @param {Boolean} isBulk - Whether the storage is for bulk data
  * @param {Boolean} persistQueue - Whether to persist the queue until processed
+ * @param {varies} customStorageMethod - Storage methods provided by the user
  */
-var initStorage = function(userPath, storageType, isBulk = false, persistQueue = false) {
+var initStorage = function(userPath, storageType, isBulk = false, persistQueue = false, customStorageMethod = null) {
     if (storageType === StorageTypes.MEMORY) {
         storageMethod = memoryStorage;
     }
+    else if (storageType === StorageTypes.CUSTOM) {
+        if (customStorageMethod) {
+            storageMethod = customStorageMethod;
```

Comment:
is there any validation to storageMethod, especially custom one. They might do some fishy things